src/relay/libp2p_relay_server.erl

```diff
@@ -173,15 +173,15 @@ handle_info({'DOWN', _Ref, process, Pid, Reason}, #state{tid=TID, stream=Pid, ad
     lager:info("Relay session with address ~p closed with reason ~p", [Address, Reason]),
     _ = libp2p_config:remove_listener(TID, Address),
     %% add it to the banlist so we temporarily avoid trying to connect to it again
-    {noreply, banlist(Address, retry(State#state{stream=undefined, address=undefined}))};
-handle_info({stop_relay, Address}, #state{stream=Pid, address=Address, tid=TID}=State) when is_pid(Pid) ->
-    lager:warning("relay was asked to be stopped ~p ~p", [Pid, Address]),
+    {noreply, retry(banlist(Address, State#state{stream=undefined, address=undefined}))};
+handle_info({replace_relay, Address}, #state{stream=Pid, address=Address, tid=TID}=State) when is_pid(Pid) ->
+    lager:warning("relay was asked to be replaced ~p ~p", [Pid, Address]),
     _ = libp2p_config:remove_listener(TID, Address),
     catch libp2p_framed_stream:close(Pid),
     %% we likely disconnected from this relay specifically because of overload or max duration
     %% so avoid immediately re-connecting to it
-    {noreply, banlist(Address, State#state{stream=undefined, address=undefined})};
-handle_info({stop_relay, _OtherAddress}, State) ->
+    {noreply, retry(banlist(Address, State#state{stream=undefined, address=undefined}))};
```

Comment:
a call to retry was added here? intentional?

Reply:
Yes, because we want to obtain a new relay address.